avldb/src/main/scala/scorex/db/LDBFactory.scala

```diff
@@ -0,0 +1,128 @@
+package scorex.db
+
+import java.io.File
+import java.util.concurrent.locks.ReentrantReadWriteLock
+
+import org.iq80.leveldb.{DB, Range, DBFactory, DBIterator, Options, ReadOptions, Snapshot, WriteBatch, WriteOptions}
+import scorex.util.ScorexLogging
+
+import scala.collection.mutable
+import scala.util.Try
+
+case class StoreRegistry(val factory : DBFactory) extends DBFactory {
+
+  val lock = new ReentrantReadWriteLock()
```

Comment:
Please comment in the code on the motivation to use this specific type of lock.

Reply:
Multithreaded application may try to create several database at the same time.
To avoid race condition in StoreRegistry we need to synchronize access to this class.
As far as I understand, ReentrantReadWriteLock is standard way of synchronizing access to the class (replacement of Java method "synchronized" qualifier).